src/components/BuyButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState, Fragment } from "react";
+import { loadStripe } from "@stripe/stripe-js";
+// Make sure to call `loadStripe` outside of a component’s render to avoid
+// recreating the `Stripe` object on every render.
+const stripePromise = loadStripe(
+  process.env.REACT_APP_STRIPE_API_KEY as string
+);
+
+const BuyButton = () => {
+  const [error, setError] = useState<string>();
+
+  const handleClick = async () => {
+    // When the customer clicks on the button, redirect them to Checkout.
+    const stripe = await stripePromise;
+    if (!stripe) {
+      return null;
+    }
+    const transaction = await stripe.redirectToCheckout({
+      lineItems: [
+        { price: process.env.REACT_APP_STRIPE_PRICE_ID as string, quantity: 1 },
```

Comment:
We should try to avoid `as` as much as possible. As it basically says to typescript "don't worry, this is a string".
However it could be `undefined`.
A better (though still not ideal) solution here would be to write `${process.env.REACT_APP_STRIPE_PRICE_ID}` which actually forces it to be a string. So even if the env var is `undefined` it will turn it into the string `"undefined"`.

Reply:
Cool, makes sense. I've converted all other references of `as` to string literals too.